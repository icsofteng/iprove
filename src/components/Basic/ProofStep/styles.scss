.lineNumber {
  display: inline-block;
  text-align: left;
  font-size: 12pt;
  padding: 10pt;
  vertical-align: middle;
}
.step {
  display: flex;
}
.latex {
  display:none;
}
.showLatex {
  display: flex;
  font-size:14px;
  flex:1;
}
.proofStep {
  flex: 1;
}