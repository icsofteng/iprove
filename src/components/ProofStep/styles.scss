.lineNumber {
  display: inline-block;
  text-align: left;
  font-size: 12pt;
  padding: 10pt;
  vertical-align: middle;
}
.step {
  display: flex;
}