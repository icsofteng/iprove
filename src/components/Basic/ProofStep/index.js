import React, { Component } from 'react'
import Latex from 'react-latex'
import cx from 'classnames'
import { translate_rule } from '../../../translator/latex'
import Rule from '../Rule'
import DependencyList from '../DependencyList'
import styles from './styles.scss'

export default class ProofStep extends Component {
  constructor(props) {
    super(props)
    this.state = { latex: false }
  }

  render() {
    const { step, index, showDependencies, offset, type, z3 } = this.props

    const isCorrect =
      (type !== 'givens' && type !== 'lemmas' && z3 === 'unsat') ||
      ['assume', 'arbitrary', 'exit', 'case'].includes(step.ast.type)

    const isError =
      this.props.raw !== '' &&
      (type !== 'givens' || (type === 'givens' && this.props.errors)) &&
      (type !== 'lemmas' || (type === 'lemmas' && this.props.errors)) &&
      z3 !== 'unsat' &&
      !isCorrect

    return (
      <React.Fragment>
        <div className={cx("proof-line", "proof-line-short", { correct: isCorrect, error: isError })}>
          { type !== 'goal' && <div className="proof-linenumber">{offset + index + 1}</div> }
          <div className={cx(styles.latex, {[styles.showLatex]: this.state.latex})}>
            <Latex>{"$"+translate_rule(step.ast)+"$"}</Latex>
          </div>
          {
            !this.state.latex &&
            <div className={styles.proofStep}>
              <Rule key={"rule" + index} {...step.ast} path={[type, index, "ast"]} />
            </div>
          }
          <button className="toggle" onClick={()=>this.setState(state => ({ latex: !state.latex }))}>Toggle</button>
        { showDependencies && <DependencyList index={index} dependencies={step.dependencies} path={[type, index, "dependencies"]} /> }
        </div>
      </React.Fragment>
    )
  }
}
