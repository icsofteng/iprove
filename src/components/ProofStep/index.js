import React from 'react'
import Rule from '../Rule'
import styles from './styles.scss'

const ProofStep = (props) =>
  <div className={styles.steps}>
    <div className={styles.lineNumber}>{props.index}</div>
    <Rule key={"rule"+props.index} {...props.rule} path={[props.index]} />
  </div>

export default ProofStep