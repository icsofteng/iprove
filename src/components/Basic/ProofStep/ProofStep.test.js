import React from 'react'
import ProofStep from './'

const ast = { type: "binary", symbol: "implies", lhs: {type:"literal", value:"p"}, rhs:  {type:"literal", value:"q"}}
const step = { dependencies: [], ast }
const wrapper = shallow(<ProofStep step={step} index={0} showDependencies={true} offset={0} type="steps" />)

test('Snapshot test', () => {
  expect(wrapper).toMatchSnapshot()
})

