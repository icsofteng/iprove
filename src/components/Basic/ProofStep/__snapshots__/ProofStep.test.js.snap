// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test 1`] = `
<Fragment>
  <div
    className="proof-line proof-line-short error"
  >
    <div
      className="proof-linenumber"
    >
      1
    </div>
    <div
      className="latex"
    >
      <Latex
        displayMode={false}
      >
        $p \\Longrightarrow q$
      </Latex>
    </div>
    <div
      className="proofStep"
    >
      <Connect(Rule)
        key="rule0"
        lhs={
          Object {
            "type": "literal",
            "value": "p",
          }
        }
        path={
          Array [
            "steps",
            0,
            "ast",
          ]
        }
        rhs={
          Object {
            "type": "literal",
            "value": "q",
          }
        }
        symbol="implies"
        type="binary"
      />
    </div>
    <button
      className="toggle"
      onClick={[Function]}
    >
      Toggle
    </button>
    <Connect(DependencyList)
      dependencies={Array []}
      index={0}
      path={
        Array [
          "steps",
          0,
          "dependencies",
        ]
      }
    />
  </div>
</Fragment>
`;
